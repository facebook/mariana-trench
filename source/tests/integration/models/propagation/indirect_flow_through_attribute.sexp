(method (private) "LFlow;.into_attribute:()V"
 (
  (load-param v0)

  LINE(1)
  (invoke-virtual (v1) "LSource;.source:()LData;")
  (move-result-object v1)

  LINE(2)
  (iput-object v1 v0 "FLow;.field:LData;")

  (return-void)
 )
)
(method (private) "LFlow;.attribute_into_sink:(LSink;LSource;)V"
 (
  (load-param v0)

  LINE(1)
  (invoke-virtual (v0) "LFlow;.into_attribute:()V")

  LINE(2)
  (invoke-virtual (v0 v0) "LSink;.sink:(LData;)V")

  (return-void)
 )
)
