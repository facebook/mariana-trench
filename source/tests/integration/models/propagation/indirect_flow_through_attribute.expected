{
  "metadata" :
  {
    "@generated" : true
  },
  "models" :
  [
    {
      "generations" :
      [
        {
          "call_position" :
          {
            "line" : 1,
            "path" : "Flow.java"
          },
          "callee" : "LFlow;.into_attribute:()V",
          "callee_port" : "Argument(0).field",
          "caller_port" : "Argument(0).field",
          "distance" : 2,
          "kind" : "TestSource",
          "origins" :
          [
            "LSource;.source:()LData;"
          ]
        }
      ],
      "issues" :
      [
        {
          "position" :
          {
            "line" : 2,
            "path" : "Flow.java"
          },
          "rule" : 1,
          "sinks" :
          [
            {
              "call_position" :
              {
                "line" : 2,
                "path" : "Flow.java"
              },
              "callee" : "LSink;.sink:(LData;)V",
              "callee_port" : "Argument(1)",
              "distance" : 1,
              "kind" : "TestSink",
              "origins" :
              [
                "LSink;.sink:(LData;)V"
              ]
            }
          ],
          "sources" :
          [
            {
              "call_position" :
              {
                "line" : 1,
                "path" : "Flow.java"
              },
              "callee" : "LFlow;.into_attribute:()V",
              "callee_port" : "Argument(0).field",
              "distance" : 2,
              "kind" : "TestSource",
              "origins" :
              [
                "LSource;.source:()LData;"
              ]
            }
          ]
        }
      ],
      "method" : "LFlow;.attribute_into_sink:(LSink;LSource;)V",
      "position" :
      {
        "line" : 0,
        "path" : "Flow.java"
      },
      "sinks" :
      [
        {
          "call_position" :
          {
            "line" : 2,
            "path" : "Flow.java"
          },
          "callee" : "LSink;.sink:(LData;)V",
          "callee_port" : "Argument(1)",
          "caller_port" : "Argument(0)",
          "distance" : 1,
          "kind" : "TestSink",
          "origins" :
          [
            "LSink;.sink:(LData;)V"
          ]
        }
      ]
    },
    {
      "generations" :
      [
        {
          "call_position" :
          {
            "line" : 1,
            "path" : "Flow.java"
          },
          "callee" : "LSource;.source:()LData;",
          "callee_port" : "Return",
          "caller_port" : "Argument(0).field",
          "distance" : 1,
          "kind" : "TestSource",
          "local_positions" :
          [
            {
              "line" : 2
            }
          ],
          "origins" :
          [
            "LSource;.source:()LData;"
          ]
        }
      ],
      "method" : "LFlow;.into_attribute:()V",
      "position" :
      {
        "line" : 0,
        "path" : "Flow.java"
      }
    }
  ]
}
